src/types.ts

```diff
@@ -16,68 +16,73 @@ export type MethodDecorator<T extends Function = any> = (
 ) => TypedPropertyDescriptor<T> | void;
 
 export interface HttpEndpointDecoratorConfig<
-    S extends TSchema = TSchema,
-    ResponseConfig extends ResponseValidatorConfig<S> = ResponseValidatorConfig<S>,
+    TTSchema extends TSchema = TSchema,
+    ResponseConfig extends ResponseValidatorConfig<TTSchema> = ResponseValidatorConfig<TTSchema>,
     RequestConfigs extends RequestValidatorConfig[] = RequestValidatorConfig[],
 > extends Omit<ApiOperationOptions, 'requestBody' | 'parameters'> {
     method: 'GET' | 'POST' | 'PATCH' | 'DELETE' | 'PUT';
     responseCode?: number;
     path?: string;
-    validate?: ValidatorConfig<S, ResponseConfig, RequestConfigs>;
+    validate?: ValidatorConfig<TTSchema, ResponseConfig, RequestConfigs>;
 }
 
-export interface SchemaValidator<T extends TSchema = TSchema> {
-    schema: T;
+export interface SchemaValidator<TTSchema extends TSchema = TSchema> {
+    schema: TTSchema;
     name: string;
-    check: TypeCheck<T>['Check'];
-    validate(data: Obj | Obj[]): Static<T>;
+    check: ValidateFunction<Static<TTSchema>>;
+    validate(data: Obj | Obj[]): Static<TTSchema>;
 }
-export interface ValidatorConfigBase {
-    schema?: TSchema;
+export interface ValidatorConfigBase<TTSchema extends TSchema> {
+    schema?: TTSchema;
     coerceTypes?: boolean;
     stripUnknownProps?: boolean;
     name?: string;
     required?: boolean;
     pipes?: (PipeTransform | Type<PipeTransform>)[];
 }
-export interface ResponseValidatorConfig<T extends TSchema = TSchema> extends ValidatorConfigBase {
-    schema: T;
+export interface ResponseValidatorConfig<TTSchema extends TSchema = TSchema> extends ValidatorConfigBase<TTSchema> {
```

Comment:
If it's easy, remove the default generic
```suggestion
export interface ResponseValidatorConfig<TTSchema extends TSchema> extends ValidatorConfigBase<TTSchema> {
```